### PR TITLE
Use the tycho-apitools-plugin to generate the api description file

### PR DESCRIPTION
**Pull request by laeubi**
The `tycho-apitools-plugin` provides a mojo now to generate the description file, this should be faster and easier to use/understand than the ANT calls.

Lets see if this causes any issues...

### CODE REVIEW

**Comment by laeubi** [comment]
This requires
- https://github.com/eclipse-tycho/tycho/pull/1759

File: eclipse-platform-parent/pom.xml [hunk 1]
```diff
@@ -745,87 +745,23 @@
...
+	            <version>${tycho.version}</version>
+				<executions>
+					<execution>
+						<id>generate</id>
+						<goals>
+							<goal>generate</goal>
+						</goals>
+						<configuration>
+                  			<skip>${skipAPIDescription}</skip>
+                  			<projectName>${project.artifactId}_${qualifiedVersion}</projectName>
```
Comment (HannesWell) [inline eclipse-platform-parent/pom.xml:760]: Shouldn't it be possible to derive that value in the mojo from the project's metadata?

**Review by HannesWell — APPROVED** [review]
Yes fully agree, that using the mojo is simpler to understand/configure and if it is faster, even better!

**Comment by akurtakov** [comment]
Jenkins build failed with :
```
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-apitools-plugin:4.0.0-SNAPSHOT:generate (generate) on project org.eclipse.osgi: Execution generate of goal org.eclipse.tycho:tycho-apitools-plugin:4.0.0-SNAPSHOT:generate failed: A required class was missing while executing org.eclipse.tycho:tycho-apitools-plugin:4.0.0-SNAPSHOT:generate: org/objectweb/asm/ClassVisitor
```
so this seems to need more work.

File: eclipse-platform-parent/pom.xml (hunk 1 above)
Comment (laeubi) [reply eclipse-platform-parent/pom.xml:760]: Actually it is derived, but as always platform uses a "special" value here (that diverges from the maven version) so to not cause any "confusion" ...

**Comment by laeubi** [comment]
> Jenkins build failed with :
> 
> ```
> [ERROR] Failed to execute goal org.eclipse.tycho:tycho-apitools-plugin:4.0.0-SNAPSHOT:generate (generate) on project org.eclipse.osgi: Execution generate of goal org.eclipse.tycho:tycho-apitools-plugin:4.0.0-SNAPSHOT:generate failed: A required class was missing while executing org.eclipse.tycho:tycho-apitools-plugin:4.0.0-SNAPSHOT:generate: org/objectweb/asm/ClassVisitor
> ```
> 
> so this seems to need more work.

Yes I already submitted a fix to tycho so lets see if that helps, just restarted the build...

**Comment by akurtakov** [comment]
They are https://repo.eclipse.org/content/repositories/eclipse/org/eclipse/eclipse-platform-parent/4.27.0-SNAPSHOT/eclipse-platform-parent-4.27.0-20221201.162228-49.pom .

**Comment by laeubi** [comment]
It seems the build is running now and I see executions of the new api-mojo, so if the build succeeds and now one has concerns this is ready to be merged.

**Comment by HannesWell** [comment]
> It seems the build is running now and I see executions of the new api-mojo, so if the build succeeds and now one has concerns this is ready to be merged.

Do you know if there is a p2-repo produced in the verification build? If yes I think it would be good to replay a run, where the repo is archived and then to check some jars if the contained api-description is fine (compared to the current one from the latest I-build).
If that is not feasible easily at least the next I-build repo should be checked for proper api-descriptions.

But besides that I have no concerns to proceed. :)

**Comment by laeubi** [comment]
> Do you know if there is a p2-repo produced in the verification build? If yes I think it would be good to replay a run, where the repo is archived and then to check some jars if the contained api-description is fine (compared to the current one from the latest I-build).

Don't think so, but the log should "complain" if there are items replaced / different to the baseline, that was why I discovered that I need a different configuration from the defaults because aggregator uses a different version string.

By the way this seem to speed up the build from ~1hr to ~45 minutes ... @akurtakov I'll leave it up to your decision to merge this, the build is green so from my side ready to go and "just try it".

**Review by HannesWell — APPROVED** [review]
> > Do you know if there is a p2-repo produced in the verification build? If yes I think it would be good to replay a run, where the repo is archived and then to check some jars if the contained api-description is fine (compared to the current one from the latest I-build).
> 
> Don't think so, but the log should "complain" if there are items replaced / different to the baseline, that was why I discovered that I need a different configuration from the defaults because aggregator uses a different version string.

Indeed the p2-baseline comparator should notice any difference. Im fine with this Change👍🏽

**Comment by mickaelistria** [comment]
Build is currentlyf failing for other reasons. Please wait for a good I-Build before merging it.

**Comment by mickaelistria** [comment]
Last build is successful, let's do it now.

**Comment by vogella** [comment]
If we do not use the api generation Ant tasks anymore in platform, we should IMHO also deprecate them in PDE or even mark them for deletion (in > 2 years we would be able to delete them, haha).

**Comment by laeubi** [comment]
Feel free to start an initiative :-)
But last time I was told that ant is still used alot I think @iloveeclipse mentioned this

**Pull request merged**